packages/client/screens/MySettings/index.tsx

```diff
@@ -0,0 +1,198 @@
+                    visibility: 'PUBLIC',
+                });
+            },
+        },
+        {
+            text: 'Followers only',
+            selected: state.hasTag('Followers Only Visibility'),
+            onPress: () => {
+                send({
+                    type: 'Update Visibility',
+                    visibility: 'FOLLOWERS_ONLY',
+                });
+            },
+        },
+        {
+            text: 'Private',
+            selected: state.hasTag('Private Visibility'),
+            onPress: () => {
+                send({
+                    type: 'Update Visibility',
+                    visibility: 'PRIVATE',
+                });
+            },
+        },
+    ];
+
+    return (
+        <View>
+            <Text sx={{ color: 'white', fontSize: 'l' }}>{title}</Text>
+
+            <View
+                sx={{
+                    flexDirection: 'row',
+                    justifyContent: 'space-around',
+                    alignItems: 'center',
+                    flexWrap: 'wrap',
+                }}
+            >
+                {options.map(({ text, selected, onPress }, index) => {
+                    return (
+                        <View
+                            key={index}
+                            sx={{
+                                paddingTop: 'xl',
+                                marginX: 'm',
+                            }}
+                        >
+                            <MtvRoomCreationFormOptionButton
+                                key={text}
+                                text={text}
+                                isSelected={selected}
+                                onPress={onPress}
+                                shouldApplyRightMargin={false}
+                            />
+                        </View>
+                    );
+                })}
+            </View>
+        </View>
+    );
+};
+
+const MySettings: React.FC<MySettingsScreenProps> = ({ navigation }) => {
+    const insets = useSafeAreaInsets();
+    const [state] = useMachine(settingsMachine);
+
+    const settings = [
+        {
+            containerTestID: 'playlists-visibility-radio-group',
+            title: 'Playlists visibility',
+            visibilitySettingActor: state.children[
+                'Playlists Visibility Manager Machine'
+            ] as VisibilitySettingMachineActor,
+        },
+        {
+            containerTestID: 'playlists-relations-radio-group',
+            title: 'Relations visibility',
+            visibilitySettingActor: state.children[
+                'Relations Visibility Manager Machine'
+            ] as VisibilitySettingMachineActor,
+        },
+        {
+            containerTestID: 'playlists-devices-radio-group',
+            title: 'Devices visibility',
+            visibilitySettingActor: state.children[
+                'Devices Visibility Manager Machine'
+            ] as VisibilitySettingMachineActor,
+        },
+    ];
```

Comment:
I think there's some experiments to be done using only selector on related actors, inside and outside an array.
I have no clue on how and when it would rerender :o